plugins/woocommerce/includes/class-wc-frontend-scripts.php

```diff
@@ -431,6 +431,12 @@ public static function load_scripts() {
 		// Global frontend scripts.
 		self::enqueue_script( 'woocommerce' );
 
+		// Only enqueue the cart fragment script if a theme/plugin is using it to update custom fragments.
+		// The cart widget handles the enqueue internally when needed, so no global enqueue is needed by default.
+		if ( has_filter( 'woocommerce_add_to_cart_fragments' ) || has_filter( 'add_to_cart_fragments' ) ) {
+			wp_enqueue_script( 'wc-cart-fragments' );
+		}
```

Comment:
Good mitigation, however, the registration of the filter would have to happen prior to the `enqueue_scripts` WP hook execution. I'm assuming this means there is still the possibility that some plugins would need to adjust where they add their filter callback?

Reply:
Yes, the original guidance to call `wp_enqueue_script` manually would apply in that scenario.